cons_results/imputation/post_imputation.py

```diff
@@ -0,0 +1,308 @@
+import warnings
+from typing import List
+
+import numpy as np
+import pandas as pd
+
+# Case 1 - all responses - calculate total (even if given or not this will be same)
+# Case 2 - some non responses - no total, impute non responses and calculate total
+# Case 3 - some non responses - given total - impute non responses and re weight imputed
+# case 4 - all non responses - no total, impute non responses and calculate total
+
+# Can the total be imputed? my guess would be no
+
+
+def calculate_totals(df: pd.DataFrame, derive_from: List[int]) -> pd.DataFrame:
+
+    """
+    Returns the sums of a dataframe in which the first level index is in
+    derive_from. The sum is based on common indices. Columns must contain float
+    or int.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Dataframe to sum, first level index must contain values from derive_from
+
+    derive_from : List[int]
+        Values to take a subset of df.
+
+    Returns
+    -------
+    sums : pd.DataFrame
+        A dataframe with sums, constain marker, and columns from index which the
+        sum was based on.
+    """
+
+    df_temp = df.fillna(0)
+
+    sums = sum(
+        [
+            df_temp.loc[question_no]
+            for question_no in derive_from
+            if question_no in df_temp.index
+        ]
+    )
+    sums.rename(columns={"target": "derived_target"}, inplace=True)
+    return sums.assign(constrain_marker=f"sum{derive_from}").reset_index()
+
+
+def create_derive_map():
+    """
+    Function to create derive mapping dictionary
+    Will check the unique values for form types and remove this
+    from the dictionary if not present. handles error
+
+    Returns
+    -------
+    dict
+        Derived question mapping in a dictionary.
+        Removes form IDs which are not present in dataframe
+    """
+    warnings.warn(
+        "This might need to be expanded to allow for multiple derived questions "
+        "and updated to correct question numbers"
+    )
+    derive_map = {
+        "derive": 5,
+        "from": [1, 2, 3, 4],
+    }
+    return derive_map
+
+
+def post_imputation_processing(
```

Comment:
While not in pep8  as far as i know function typically should start with verb
```suggestion
def process_post_imputation(
```